setup.py

```diff
@@ -26,7 +26,7 @@
     long_description=long_description,
     long_description_content_type='text/markdown',
     packages=find_packages(),
-    install_requires=['numpy', 'tensorflow', 'six'],
+    install_requires=['numpy', 'tensorflow', 'six', 'graphviz', 'ipython'],
```

Comment:
Would it be much trouble to not have these as a hard dependency?

Reply:
I would agree. `ipython` isn't much of a concern, but `graphviz` can be difficult to install on some platforms. I put a comment into `graphviz_wrapper.py` with instructions for making the dependency on `graphviz` a soft dependency.

Reply:
Indeed, I wasn't sure if it is good idea to make graphviz as a hard dependency.
Updated code to load it lazily.